torch/fx/graph_module.py

```diff
@@ -704,12 +704,15 @@ def __reduce__(self):
     # we need to define deepcopy otherwise it will call __reduce__
     # and cause symbolic tracing to occur every time we try to copy the object
     def __deepcopy__(self, memo):
+        res = GraphModule(self, self.graph)
+        memo[id(self)] = res
         fake_mod = torch.nn.Module()
-        fake_mod.__dict__ = copy.deepcopy(self.__dict__)
-        res = GraphModule(fake_mod, fake_mod.__dict__['_graph'])
-        res.meta = copy.deepcopy(getattr(self, 'meta', {}))
+        fake_mod.__dict__ = copy.deepcopy(self.__dict__, memo)
+        GraphModule.__init__(res, fake_mod, fake_mod.__dict__['_graph'])
+        res.meta = copy.deepcopy(getattr(self, 'meta', {}), memo)
```

Comment:
I think if you just pass `memo` into all of the deepcopy calls, it should would work without having to mess with `memo` itself? (i.e. I think copy.deepcopy implementation is supposed to be the thing that manages the `memo` object)

i.e.

```python
fake_mod = torch.nn.Module()
fake_mod.__dict__ = copy.deepcopy(self.__dict__, memo)
res = GraphModule(fake_mod, fake_mod.__dict__['_graph'])
res.meta = copy.deepcopy(getattr(self, 'meta', {}), memo)
```

also btw, is it intended to be `fake_mod.__dict__ = copy.deepcopy(self.__dict__)` or should it be `fake_mod.__dict__ = copy.deepcopy(self.root.__dict__)` ?

Reply:
self.root doesnt exist: we didnt store root (maybe we should)...